src/formatDate.js

```diff
@@ -50,7 +50,29 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const fromSeparator = fromFormat[3];
+  const toSeparator = toFormat[3];
+  const oldDate = date.split(fromSeparator);
+  const newDate = [];
+  const dateInfo = {};
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
+    dateInfo[fromFormat[i]] = oldDate[i];
+  }
+
+  if (!fromFormat.includes('YY')) {
+    dateInfo.YY = dateInfo.YYYY.slice(2);
+  } else {
+    dateInfo.YYYY = dateInfo.YY < 30
+      ? 20 + dateInfo.YY
+      : 19 + dateInfo.YY;
+  }
```

Comment:
It can be a part of a loop. Here inside include() you create one more loop, but it's an expensive operation.